os/net/mac/tsch/sixtop/sixp.c

```diff
@@ -190,16 +232,35 @@ sixp_input(const uint8_t *buf, uint16_t len, const linkaddr_t *src_addr)
 
   if(pkt.type == SIXP_PKT_TYPE_REQUEST) {
     if(trans != NULL) {
-      /* Error: not supposed to have another transaction with the peer. */
-      LOG_ERR("6P: sixp_input() fails because another request [peer_addr:");
-      LOG_ERR_LLADDR((const linkaddr_t *)src_addr);
-      LOG_ERR_(" seqno:%u] is in process\n", sixp_trans_get_seqno(trans));
-      if(send_back_error(SIXP_PKT_TYPE_RESPONSE,
-                         (sixp_pkt_code_t)(uint8_t)SIXP_PKT_RC_ERR_BUSY,
-                         pkt.sfid, pkt.seqno, src_addr) < 0) {
-        LOG_ERR("6P: sixp_input() fails to return an error response");
+      if(pkt.code.cmd != SIXP_PKT_CMD_CLEAR &&
+         ((pkt.seqno == 0 && sixp_trans_get_seqno(trans) != 0) ||
+          (pkt.seqno != 0 && sixp_trans_get_seqno(trans) == 0))) {
+        /*
+         * seems the peer had power-cycle; we're going to send back
+         * RC_ERR_SEQNUM. in this case, we don't want to allocate
+         * another transaction for this new request.
+         */
+        goto err_seqnum;
```

Comment:
Jumping into another conditional branch looks tricky to me. I would refactor the procedure under `err_seqnum` label into a function, and call it at the two places.

Reply:
I take your comment.